### PR TITLE
Expose snap point

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

Expose projectToPoly as `snap_point`.  Sometimes it is useful to be able to snap a point to the nearest navigable point, people had been using `pathfinder.try_step(pt, pt)` to do that, but this has some weird edge cases.

## How Has This Been Tested

Via the test!  If you swap `snap_point(pt)` for `try_step(pt, pt)`, it finds a failure very fast.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: tests/test_snap_point.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@pytest.fixture(scope="function")
+def test_data():
+    pf = habitat_sim.PathFinder()
+    pf.load_nav_mesh(
+        "data/scene_datasets/habitat-test-scenes/skokloster-castle.navmesh"
+    )
+
+    return pf, pf.get_random_navigable_point()
```
Comment (msbaines) [inline tests/test_snap_point.py]: Will this be deterministic?

Comment (erikwijmans) [reply tests/test_snap_point.py]: It's not (seeding the pathfinder is pretty much impossible due to recast using just raw `rand()` and `srand` only kinda working).

I have tried a number of invocations and let hypothesis run out to 100k examples and it couldn't find anything.

Comment (msbaines) [reply tests/test_snap_point.py]: Maybe add a small comment saying that this isn't deterministic.

Comment (erikwijmans) [reply tests/test_snap_point.py]: Done
